apps/mosaico/src/Mosaico/Footer.purs

```diff
@@ -1,61 +1,295 @@
         ]
     }
   where
-    externalLink caption url =
-      DOM.a
-        { href: url
-        , children: [ DOM.text caption ]
-        }
-    footerLink caption link =
-      DOM.a
-        { href: "/sida/" <> link
-        , children: [ DOM.text caption ]
-        , onClick: onStaticPageClick link
-        }
-    logo :: Paper -> JSX
-    logo paper =
-      let className = "mosaico-footer__logo mosaico-footer__logo--" <> cssName paper
-          children =
-            [ DOM.div { className: "mosaico-footer__logo-image mosaico-footer__logo-image--" <> cssName paper }
-            , DOM.div
-                { className: "mosaico-footer__logo-caption"
-                , children: [ DOM.text $ paperName paper ]
-                }
-            ]
-      in if paper == mosaicoPaper
-         then DOM.div { className, children }
-         else DOM.a { className, children, href: homepage paper }
+  logo :: Paper -> JSX
+  logo paper =
+    let
+      className = "mosaico-footer__logo mosaico-footer__logo--" <> cssName paper
+
+      children =
+        [ DOM.div { className: "mosaico-footer__logo-image mosaico-footer__logo-image--" <> cssName paper }
+        , DOM.div
+            { className: "mosaico-footer__logo-caption"
+            , children: [ DOM.text $ paperName paper ]
+            }
+        ]
+    in
+      if paper == mosaicoPaper then
+        DOM.div { className, children }
+      else
+        DOM.a { className, children, href: homepage paper }
+
+contactInfo :: Paper -> (String -> EventHandler) -> JSX
+contactInfo ON = ostnylandContactInfo
+
+contactInfo VN = vastranylandContactInfo
+
+contactInfo HBL = hblContactInfo
+
+contactInfo _ = footerLinks
+
+hblContactInfo :: (String -> EventHandler) -> JSX
+hblContactInfo onStaticPageClick =
+  DOM.div
+    { className: "mosaico-footer__contact-info"
+    , children:
+        [ DOM.div
+            { className: "mosaico-footer__contact-info__columns"
+            , children:
+                [ column firstColumn
+                , column secondColumn
+                , column thirdColumn
+                ]
+            }
+        , footerLinks onStaticPageClick
+        ]
+    }
+  where
+  firstColumn =
+    [ section "Besöksadress"
+        [ DOM.text "Mannerheimvägen 18"
+        , DOM.br {}
+        , DOM.text "00100 Helsingfors"
+        ]
+    , section "Växel: " [ tel "09 125 31" ]
+    , section "E-post: " [ DOM.text "fornamn.efternamn@hbl.fi" ]
+    , section "Prenumerationer och Kundservice: "
+        [ tel "09 125 35 00"
+        , DOM.text ", "
+        , email "pren@ksfmedia.fi"
+        , DOM.br {}
+        , DOM.text "(mån-fre kl. 8.00-12.00 och 13.00-16.00."
+        , DOM.text "På onsdagar stänger kundservice kl. 15.00)"
+        ]
+    , section "Annonser: "
+        [ tel "09 125 35 58"
+        , DOM.text ", "
+        , email "annons@hbl.fi"
+        ]
+    ]
+
+  secondColumn =
+    [ section "Erja Yläjärvi" [ DOM.text "Chefredaktör och ansvarig utgivare" ]
```

Comment:
Let's add a whitespace between the title and the content, as in narrower screens these joined together.

Maybe the responsibility of doing so could go to the `section` function even.